packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -0,0 +1,376 @@
+    super(scope, id, props);
+    this.integrationPattern = props.integrationPattern || sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, EcsRunTask.SUPPORTED_INTEGRATION_PATTERNS);
+
+    if (this.integrationPattern === sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN && !sfn.FieldUtils.containsTaskToken(props.containerOverrides)) {
+      throw new Error('Task Token is required in `containerOverrides` for callback. Use Context.taskToken to set the token.');
+    }
+
+    this.validateLaunchTarget();
+
+    for (const override of this.props.containerOverrides || []) {
+      const name = override.containerDefinition.containerName;
+      if (!cdk.Token.isUnresolved(name)) {
+        const cont = this.props.taskDefinition.node.tryFindChild(name);
+        if (!cont) {
+          throw new Error(`Overrides mention container with name '${name}', but no such container in task definition`);
+        }
+      }
+    }
+
+    this.taskPolicies = this.makePolicyStatements();
+  }
+
+  protected renderTask(): any {
+    if (this.networkConfiguration !== undefined) {
+      // Make sure we have a security group if we're using AWSVPC networking
+      this.securityGroup = this.securityGroup ?? new ec2.SecurityGroup(this, 'SecurityGroup', { vpc: this.props.cluster.vpc });
+      this.connections.addSecurityGroup(this.securityGroup);
+    }
+
+    return {
+      Resource: integrationResourceArn('ecs', 'runTask', this.integrationPattern),
+      Parameters: sfn.FieldUtils.renderObject({
+        Cluster: this.props.cluster.clusterArn,
+        TaskDefinition: this.props.taskDefinition.taskDefinitionArn,
+        NetworkConfiguration: this.networkConfiguration,
+        Overrides: renderOverrides(this.props.containerOverrides),
+        ...this.props.launchTarget.bind(this).parameters,
+      }),
+    };
+  }
+
+  private configureAwsVpcNetworking() {
+    const subnetSelection = this.props.subnets ?? { subnetType: this.props.assignPublicIp ? ec2.SubnetType.PUBLIC : ec2.SubnetType.PRIVATE };
+
+    this.networkConfiguration = {
+      AwsvpcConfiguration: {
+        AssignPublicIp: this.props.assignPublicIp ? (this.props.assignPublicIp ? 'ENABLED' : 'DISABLED') : undefined,
+        Subnets: this.props.cluster.vpc.selectSubnets(subnetSelection).subnetIds,
+        SecurityGroups: cdk.Lazy.listValue({ produce: () => [this.securityGroup!.securityGroupId] }),
+      },
+    };
+  }
+
+  private validateLaunchTarget() {
+    if (this.props.launchTarget instanceof EcsFargateLaunchTarget) {
+      this.validateFargateLaunchType();
+    }
+
+    if (this.props.launchTarget instanceof EcsEc2LaunchTarget) {
+      this.validateEc2LaunchType();
+    }
+
+    if (!this.props.taskDefinition.defaultContainer) {
+      throw new Error('A TaskDefinition must have at least one essential container');
+    }
+  }
+
+  private validateEc2LaunchType() {
+    if (!this.props.cluster.hasEc2Capacity) {
```

Comment:
Looks like taskDefinition has a `isEc2Compatible`. Does this also need to be checked or does this one cover that as well?

Reply:
good catch, it was there and  i must have lost it =/